packages/@react-native-windows/cli/src/runWindows/utils/vstools.ts

```diff
@@ -156,25 +156,27 @@ export function addProjectToSolution(
     '\tGlobalSection(SolutionConfigurationPlatforms) = preSolution',
     '\tEndGlobalSection',
     false,
-  ).map(line => line.match(/\s+([\w|]+)\s=/)![1]);
+  ).map(line => line.match(/\s+([\w\s|]+)\s=/)![1]);
 
   const projectConfigLines: string[] = [];
 
   slnConfigs.forEach(slnConfig => {
-    projectConfigLines.push(
-      `\t\t${projectGuid}.${slnConfig}.ActiveCfg = ${
-        project.projectLang === 'cpp'
-          ? slnConfig.replace('x86', 'Win32')
-          : slnConfig
-      }`,
-    );
-    projectConfigLines.push(
-      `\t\t${projectGuid}.${slnConfig}.Build.0 = ${
-        project.projectLang === 'cpp'
-          ? slnConfig.replace('x86', 'Win32')
-          : slnConfig
-      }`,
-    );
+    if (project.projectLang !== 'cpp' || !slnConfig.endsWith('|Any CPU')) {
```

Comment:
This logic seems wrong. Neither C++ nor C# UWP app projects have an "Any CPU" configuration, not in our templates, and not in vanilla new projects in VS. Only C# WinRT Components created in VS get this (which I think is a bug, since you never submit Any CPU binaries to the Store).

Anyway, all we want to do is prevent injecting a project reference into the solution with an "Any CPU" configuration. There shouldn't be such a solution config, unless the developer manually added their own C# WinRT component project using VS and thereby created the unneeded "Any CPU" config.

Anyway, I think the current logic does the opposite of what we want - it always injects the Any CPU config if the project is C#.

Shouldn't it be just `if (!slnConfig.endsWith('|Any CPU')) {`

Reply:
@jonthysell the scenario I hit this problem was with a solution that has: 
- the RNW projects
- a RNW test app
- a RNW community module
- a C# net core 3 console app
the last piece is what throws a wrench into things, as the only project config for that is Any CPU. That adds the Any CPU config to the solution itself.
thoughts on what we should do in that case?

Reply:
I think what I said still applies - autolinking is in the business of inserting C++/WinRT or C# WinRT Component projects into the solution and making sure their configs match up with the app's configs so all of the app flavors build correctly. We do this by looking at the solution configs, and the bug here is just that there's an Any CPU config that we don't want to consider.

For autolinking, technically we should loop through the app's configs and then only inject the lib's configs that match. And maybe throw an warning if the lib is missing configs because that means that app won't be able to build that config. But since (other than the Any CPU case) we build the rnw app by building the solution, I think it's fine to loop through the solution configs instead, and just filter out Any CPU if it's there.

If a user opens up the solution in VS in and build Any CPU, then all RNW projects shouldn't build as you don't want to produce Any CPU binaries for RNW apps/libs.

Now for completeness sake, if we want to this "add project to solution" utility function to completely mirror the vanilla VS behavior of adding a new project, we should do something like:

1. Get the list of the new project's configs
    1. Inject the project reference with all of its configs
2. Get the list of solution configs
    1. Inject new solution config for every project config that doesn't exist
3. Get the list of all other projects
    1. Inject new project configs for every project so the matrix is complete

But I don't think that's necessary for our purposes.

Reply:
alright, I'll just exclude 'any cpu'. I think there's still a case to be made for adding a C# WinRT component to a solution from within VS, we have customers who do that, but we can tackle that separately. Thanks for taking a look!